src/js/components/dropdown.js

```diff
@@ -1,291 +1,395 @@
+      /************************************************************************
+       * Closes the dropdown.
+       ***********************************************************************/
 
-        // Did it come from inside open disclosure?
-        if (this.targetElement.contains(event.target)) return;
+      close () {
+        if (!this._isOpen()) { return }
 
-        // If it came from outside component, close all open disclosures
-        if (!toggle && this.activeToggle !== null) {
-          this.close();
-          return;
-        }
+        if (!this._eventDispatched('dropdownClosed')) { return }
 
-        // Check which toggle the click came from, and whether it's already opened
-        if (toggle !== this.toggleElement || this.isOpen) {
-          this.close();
-        } else {
-          this.open();
-        }
+        this._setClosedState()
+      },
+
+      /************************************************************************
+       * Returns true if the dropdown is open.
+       *
+       * @private
+       * @returns {boolean} Dropdown is open
+       ***********************************************************************/
+
+      _isOpen () {
+        return this.isOpen
       },
 
-      _shouldHandleKeydown(event) {
-        // If the keydown didn't come from within disclosure component, then bail.
-        if (!this.element.contains(event.target)) return false;
-    
-        // Delegate event to only this instance of the disclosure
-        const disclosure = event.target.closest('[data-rvt-dropdown]');
-        if (disclosure !== this.element) return false;
-    
-        return true;
+      /************************************************************************
+       * Sets the dropdown's state properties to represent it being closed.
+       *
+       * @private
+       ***********************************************************************/
+
+      _setClosedState () {
+        this.toggleElement.setAttribute('aria-expanded', 'false')
+        this.menuElement.setAttribute('hidden', '')
+
+        this.isOpen = false
+      },
+
+      /************************************************************************
+       * Returns true if the custom event with the given name was successfully
+       * dispatched.
+       *
+       * @private
+       * @param {string} name - Event name
+       * @returns {boolean} Event successfully dispatched
+       ***********************************************************************/
+
+      _eventDispatched (name) {
+        const dispatched = Component.dispatchCustomEvent(name, this.element)
+
+        return dispatched
+      },
+
+      /************************************************************************
+       * Handles click events broadcast to the disclosure.
+       *
+       * @param {Event} event - Click event
+       ***********************************************************************/
+
+      onClick (event) {
```

Comment:
We discussed this face-to-face, but we need to make sure we add back in the functionality that closes the dropdown menu when the click came from outside the dropdown menu itself. So if you say click on the document body, or another adjacent dropdown toggle (like in the header) it should close the currently open dropdown.